src/CommandLine.cc

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "CommandLine.h"
+
+static void
+ResetGetopt(const bool allowStderrWarnings)
+{
+    opterr = allowStderrWarnings ? 1 : 0;
```

Comment:
Why not just write `opterr = allowStderrWarnings;` ? The C++ allows this:

> If the source type is bool, the value false is converted to zero and the value true is converted to one

(taken from a StackOverflow [answer](https://stackoverflow.com/questions/5369770/bool-to-int-conversion)).

Reply:
Your call. I am a bit paranoid when it comes to C-C++ boundaries, but that is just paranoia here -- I see no reason why your simpler version would not work.

Reply:
Done.